File: MassMarket.Web/src/app/models/search-model.ts
export class SearchModel {
    queryText = null;
    categoryId = 0;
    ordering = 0;
    pageNr = 1;
    priceRange: number[] = [];
    metaFieldOptions: number[] = [];

    constructor(searchModel: SearchModel) {
        if (searchModel) {
            this.queryText = searchModel.queryText;
            this.categoryId = searchModel.categoryId;
            this.ordering = searchModel.ordering;
            this.pageNr = searchModel.pageNr;
            this.priceRange = searchModel.priceRange;
            this.metaFieldOptions = searchModel.metaFieldOptions;
        }
    }
}

File: MassMarket.Service/Models/SearchModel.cs
using System.Linq;
using MassMarket.Service.Helpers;

namespace MassMarket.Service.Models {

    public class SearchModel {

        public string QueryText { get; set; }
        public int CategoryId { get; set; }
        public int[] PriceRange { get; set; }
        public ProductOrdering Ordering { get; set; }
        public int PageNr { get; set; }

        public int? MinPrice {
            get {
                if (PriceRange.Length == 2) {
                    return PriceRange[0];
                }

                return null;
            }
        }

        public int? MaxPrice {
            get {
                if (PriceRange.Length == 2) {
                    return PriceRange[1];
                }

                return null;
            }
        }
    }
}
